lua/cmp/view/docs_view.lua

```diff
@@ -47,12 +53,14 @@ docs_view.open = function(self, e, view)
     self.entry = e
     vim.api.nvim_buf_call(self.window:get_buffer(), function()
       vim.cmd([[syntax clear]])
-      vim.api.nvim_buf_set_lines(self.window:get_buffer(), 0, -1, false, {})
+      vim.api.nvim_buf_set_option(0, 'modifiable', true)
+      vim.api.nvim_buf_set_lines(0, 0, -1, false, {})
+      vim.lsp.util.stylize_markdown(0, documents, {
+        max_width = max_width,
+        max_height = documentation.max_height,
+      })
+      vim.api.nvim_buf_set_option(0, 'modifiable', false)
     end)
-    vim.lsp.util.stylize_markdown(self.window:get_buffer(), documents, {
-      max_width = max_width,
-      max_height = documentation.max_height,
-    })
```

Comment:
This change was primarily needed because we need to toggle the `modifiable` buffer option while settings its contents. Since there's already a call to `nvim_buf_call` I figured all buffer modifications could be moved inside that callback